easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -607,11 +608,56 @@ def dependencies(self):
         # each processed easyconfig to remove the unwanted dependencies
         self.log.debug("Dependencies BEFORE filtering: %s" % deps)
         filter_deps = build_option('filter_deps')
+
         if filter_deps:
+            # find if any of the filter_deps are in the form name=version or name=[low:high[
+            complex_fdeps_list = [x for x in filter_deps if "=" in x]
+            complex_fdeps = { }
+            # convert list of k=v into dictionary
+            if complex_fdeps_list:
+                for s in complex_fdeps_list:
+                    k,v = s.split('=')
+                    
+                    # test whether this is a range
+                    if ":" in v:
+                        # remove range characters
+                        values = v.translate(None, '][()')
```

Comment:
The range characters are actually overkill imho, especially since the Pythonic notation `low:high` (with `high` excluded) is more natural for version ranges (e.g., `some_list[2:6]`)

In addition, this will greatly simplify the code below too.

Are you also allowing open ended like `low:` and `:high`?